### PR TITLE
Layers with MC Dropout implemented for the output modules.

### LINKED ISSUE #15: PR for Monte Carlo
- [ ] Inheritance of transformer
- [ ] Doc strings
- [ ] Check for data existence and throw exception if not

### PR DESCRIPTION
**Pull request by AsenaKarolin**
Layers with Monte Carlo dropout are implemented for the output modules. Different types of MC dropout can be chosen and dropout rate can be adjusted. Example training loop is included.

### CODE REVIEW

File: examples/monte_carlo/train_mc.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+        istrainablesigma=True,
+    )
+
+    print(model)
+    print(f"Number of parameters: {sum(p.numel() for p in model.parameters())}")
+
+    epochs = 50
+    batch_size = 20
+
+    data_train, data_test = get_data_electricity()
```
Comment (oskar-taubert) [inline examples/monte_carlo/train_mc.py:178]: The script should check, if the data is there and if it is not, download it or throw and error that explains how to get it.

File: modular_transformer/classical_mc.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+from torch import Tensor
+from torch.nn import ReLU, Softmax
+
+from .base import Transformer
+from .layers import ClassicalTransformerDecoderLayer, ClassicalTransformerEncoderLayer
+from .layers.attention_modules.attention_mechanisms.masking import AttentionMatrixMask
+from .layers.attention_modules.output_modules import LinearMCDOutputModule
+
+
+class ClassicalMCDTransformer(Transformer):
```
Comment (oskar-taubert) [inline modular_transformer/classical_mc.py:13]: Why is this not inheriting from ClassicalTransformer?

File: modular_transformer/layers/attention_modules/output_modules/linear_mcd.py
Comment (oskar-taubert) [inline modular_transformer/layers/attention_modules/output_modules/linear_mcd.py:1]: why is this in attention modules?

File: modular_transformer/layers/attention_modules/output_modules/mc_dropout.py
Comment (oskar-taubert) [inline modular_transformer/layers/attention_modules/output_modules/mc_dropout.py:1]: why is this in attention_modules?

**Review by oskar-taubert — COMMENTED** [review]
Docstrings are missing.

File: modular_transformer/layers/attention_modules/output_modules/linear_mcd.py
Comment (AsenaKarolin) [reply modular_transformer/layers/attention_modules/output_modules/linear_mcd.py:1]: It is a type of output module, that's why I put it there. What would be a better place for it?

File: modular_transformer/classical_mc.py (hunk 2 above)
Comment (AsenaKarolin) [reply modular_transformer/classical_mc.py:13]: Fair question, I will change it so that it does.

File: modular_transformer/layers/attention_modules/output_modules/linear_mcd.py
Comment (oskar-taubert) [reply modular_transformer/layers/attention_modules/output_modules/linear_mcd.py:1]: I was not sure an output module would be an attention module.
I would just leave it there for now.

**Review by oskar-taubert — APPROVED** [review]
